File: Studio/src/app/data/entity/City.ts
import { Guid } from "guid-typescript";
import { BaseEntity } from "./BaseEntity";
import { Country } from "./Country";

export interface City extends BaseEntity {
    cityName: string; // CityName in C#
    countryId: Guid; // Guid represented as string
    country?: Country; // Reference to Country class, optional
    locations?: Location[];
}

File: Studio-BE/src/NM.Studio.Domain/Entities/Locations/City.cs
using NM.Studio.Domain.Entities.Bases;
using System.ComponentModel.DataAnnotations.Schema;

namespace NM.Studio.Domain.Entities.Locations
{
    [Table("City")]
    public class City : BaseEntity
    {
        [Column(TypeName = "nvarchar(255)")]
        public string CityName { get; set; }
        
        [ForeignKey("Country")]
        public Guid CountryId { get; set; }

        public virtual Country Country { get; set; }
        
        public virtual IList<Location> Locations { get; set; }
    }
}
